### PR TITLE
Update OpenAPI spec with our latest APIs

### PR DESCRIPTION
**Pull request by Nutlope**
# APIs added

- [x] Finetuning (files + fine-tune)
- [x] Function calling
- [x] Image endpoint
- [x] Models endpoint

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -135,54 +135,356 @@ paths:
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Fine-tune job details retrieved successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FinetuneResponse'
+    post:
```
Comment (orangetin) [inline openapi.yaml:435]: the cancel path is `/fine-tunes/{id}/cancel`. Ref [here](https://github.com/togethercomputer/together/blob/cc806455f3c5073dd01bdc59bade1f7873e653e8/src/together/resources/finetune.py#L154C39-L154C45)

File: openapi.yaml [hunk 2]
```diff
@@ -135,54 +135,356 @@ paths:
...
+          schema:
+            type: string
+      responses:
+        '200':
+          description: List of fine-tune events
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/FinetuneListEvents'
+  /fine-tunes/{id}/download:
```
Comment (orangetin) [inline openapi.yaml:466]: download doesn't work like this currently, there is a redirection policy set up. maybe we just leave this out for now?

File: openapi.yaml [hunk 3]
```diff
@@ -331,18 +633,46 @@ components:
...
+          description: A list of tools to be used in the query.
+          items:
+            $ref: '#/components/schemas/ToolsPart'
+        tool_choice:
+          type: object
+          description: The choice of tool to use.
+          oneOf:
+            - type: string
+              example: 'tool_name'
+            - $ref: '#/components/schemas/ToolChoice'
```
Comment (orangetin) [inline openapi.yaml:659]: I don't think we support tools/json mode for /v1/completions, right?

**Review by orangetin — CHANGES_REQUESTED** [review]
left some comments, but otherwise looks great!

File: openapi.yaml [hunk 4]
```diff
@@ -135,54 +135,356 @@ paths:
...
+      responses:
+        '200':
+          description: Image generated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ImageResponse'
+  /files:
+    post:
+      summary: Upload a file
```
Comment (orangetin) [inline openapi.yaml:281]: file upload doesn't work like this

Comment (Nutlope) [reply openapi.yaml:281]: Decided not to include file upload anymore so I'm removing it (discussed)

File: openapi.yaml (hunk 3 above)
Comment (Nutlope) [reply openapi.yaml:659]: Correct, moved to chat completions!

File: openapi.yaml (hunk 2 above)
Comment (Nutlope) [reply openapi.yaml:466]: Fixed this

File: openapi.yaml (hunk 1 above)
Comment (Nutlope) [reply openapi.yaml:435]: Added a cancel endpoint!

Comment (orangetin) [reply openapi.yaml:410]: great! let's remove lines [410-424](https://github.com/togethercomputer/openapi/pull/7/files#diff-d910ba2ef878f7db0223a966b81c8b3f3b65027bb39e4431bb05140171eece39R410-R424) now?

**Review by orangetin — CHANGES_REQUESTED** [review]
responded to the comment. One more change, can you add back the safety_model to `CompletionRequest`? safety model is supported, technically.

File: openapi.yaml (hunk 1 above)
Comment (Nutlope) [reply openapi.yaml:410]: Good callout, done!

**Comment by Nutlope** [comment]
> responded to the comment above. One more change, can you add back the safety_model to `CompletionRequest`? safety model is supported, technically.

Good catch, done!

**Review by orangetin — APPROVED** [review]

**Pull request merged**